File: src/main/webapp/src/services/api/restrictionService.ts
/**
 * Copyright (C) 2023 GIP-RECIA, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import type { AxiosResponse } from 'axios'
import type { StructureRestriction } from '@/types/index.ts'
import { instance as axios, errorHandler } from '@/utils/index.ts'

async function getRestrictions(
  id: number,
): Promise<StructureRestriction | undefined> {
  try {
    const response: AxiosResponse<StructureRestriction> = await axios.get(
      `/api/restriction/${id}`,
    )

    return response.data
  }
  catch (e) {
    errorHandler(e)
  }
}

async function saveRestrictions(
  id: number,
  member: string,
  group: boolean,
): Promise<boolean> {
  try {
    await axios.post(
      `/api/restriction/${id}`,
      {
        member,
        group,
      },
    )

    return true
  }
  catch (e) {
    errorHandler(e)

    return false
  }
}

export {
  getRestrictions,
  saveRestrictions,
}

File: src/main/java/fr/recia/glc/services/restriction/RestrictionService.java
/*
 * Copyright (C) 2023 GIP-RECIA, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package fr.recia.glc.services.restriction;

import fr.recia.glc.configuration.GLCProperties;
import fr.recia.glc.web.dto.restriction.RestrictionEtab;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpMethod;
import org.springframework.http.MediaType;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

@Service
public class RestrictionService {

    private final RestTemplate restTemplate;
    private final GLCProperties glcProperties;
    private final static String API_KEY_HEADER = "x-api-key";

    public RestrictionService(RestTemplate restTemplateRestriction, GLCProperties glcProperties) {
        this.restTemplate = restTemplateRestriction;
        this.glcProperties = glcProperties;
    }

    public RestrictionEtab getRestrictions(String uai) {
        HttpHeaders headers = new HttpHeaders();
        headers.set(API_KEY_HEADER, glcProperties.getRestrictionRentree().getApiKey());
        HttpEntity<?> entity = new HttpEntity<>(headers);
        return restTemplate.exchange(glcProperties.getRestrictionRentree().getUrl() + "/" + uai, HttpMethod.GET, entity, RestrictionEtab.class).getBody();
    }

    public void setNewRestriction(String uai, RestrictionEtab restrictionEtab) {
        HttpHeaders headers = new HttpHeaders();
        headers.set(API_KEY_HEADER, glcProperties.getRestrictionRentree().getApiKey());
        HttpEntity<?> entity = new HttpEntity<>(restrictionEtab, headers);
        restTemplate.exchange(glcProperties.getRestrictionRentree().getUrl() + "/" + uai, HttpMethod.POST, entity, RestrictionEtab.class);
    }

}
